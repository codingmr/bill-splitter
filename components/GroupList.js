import React, { Component } from 'react';
import { Text, View, FlatList, StyleSheet, Picker } from 'react-native';

import { ListItem, Icon, Input } from 'react-native-elements';

import DialogInput from 'react-native-dialog-input';

import CustomPicker from './CustomPicker';

import ItemTypePicker from './ItemTypePicker';

export default class GroupList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      count: 0,
      billTotal: 0,
      selectedGroupIndex: -1,
      selectedItemIndex: -1,
      textHolder: '',
      Group: [],
      isDialogVisible: false,
    }

  }

  newGroup() {
    this.setState({count: this.state.count + 1})
    this.groupList_Ref.scrollToEnd({animated: true });

    this.setState(prevState => {
      // set focus to first textbox in the correct group
      return { Group: [...prevState.Group, ...[{key: this.state.count, title: 'Group ' + this.state.count, groupTotal: 0, numberPersons: 1, tipPercentage: 0, billItem: [{id: 0, itemAmount: '0.00', itemIcon: 'restaurant'}]}]] }
    })

  }

  handleOnChangeText(value) {
    /// ADD BOX
    // To achieve adding only 1 new box when input is entered in to the previous one:
    //  match the number of inputboxs with the selected inputbox index

    this.setState(prevState => {
      let groupies = [...prevState.Group]
      let groupIdx = this.state.selectedGroupIndex
      let itemIdx = this.state.selectedItemIndex

      let itemies = [...groupies[groupIdx].billItem ]
      let initialValue = value

      if (isNaN(initialValue) || initialValue=='') {initialValue = 0}

      itemies[itemIdx] = {...itemies[itemIdx], itemAmount: initialValue}
      groupies[groupIdx] = {...groupies[groupIdx], billItem: itemies}

      // sum group totals into object array
      let groupTotals = groupies.map(group => ({
        key: group.key,
        groupTitle: group.title,
        groupTot: group.billItem.reduce(function (accumulator, currentValue) {
          return Math.round( ((parseFloat(accumulator) + parseFloat(currentValue.itemAmount)) + Number.EPSILON) * 100) / 100
        }, 0)
      }));
      groupies[groupIdx] = {...groupies[groupIdx], groupTotal: groupTotals[groupIdx].groupTot}

      // ADD BOX
      if (value != '' && (itemies.length-1 == itemIdx)) {

          itemies = [...itemies, {id: itemies.length, itemAmount: '0.00', itemIcon: 'restaurant'}]
          groupies[groupIdx] = {...groupies[groupIdx], billItem: itemies}
      }
      // DELETE BOX
      else if (itemies.length-1 != itemIdx
                  && (initialValue==''
                  && (itemies[itemIdx+1].itemAmount=='' || itemies[itemIdx+1].itemAmount=='0.00')
                  && itemies.length-2 == itemIdx )) {
          // if not last inputbox
          // and current input is empty
          // and next inputbox is empty
          // and selected inputbox is one from the last inputbox
          itemies.pop()
          groupies[groupIdx] = {...groupies[groupIdx], billItem: itemies}
      }
      return {Group: groupies}
    })

    // UPDATE STATE AFTER SUM TOTALS
    this.updateBillTotal()
  }

  handleDeleteGroup = (item, index) => {
    this.setState(prevState => {
      let groupies = [...prevState.Group]
      let filteredGroupies = groupies.slice(0, index).concat(groupies.slice(index + 1, groupies.length))

      return {Group: filteredGroupies}
    })
    this.setState(prevState => {
      return {count: prevState.count-1}
    })

    this.setState({selectedGroupIndex: 0})
    this.setState({selectedItemIndex: 0})

    // Update bill total
    this.updateBillTotal()
  }

    updateBillTotal(){
      this.setState(prevState => {
        let groupies = [...prevState.Group]

        let groupTotals = groupies.map(group => ({
          key: group.key,
          groupTotTip: group.tipPercentage,
          groupTot: group.groupTotal,
        }))

        let sumGroupTot = groupTotals.reduce(function (accumulator, currentValue) {
          return Math.round( ((parseFloat(accumulator) + parseFloat(currentValue.groupTot)) + Number.EPSILON) * 100) / 100
        }, 0)

        let sumGroupTotalTips = groupTotals.reduce(function (accumulator, currentValue) {
          return Math.round( ((parseFloat(accumulator) + parseFloat(currentValue.groupTot*(currentValue.groupTotTip/100))) + Number.EPSILON) * 100) / 100
        }, 0)

        this.callParentGiveSum(sumGroupTot, sumGroupTotalTips)

        return {billTotal: sumGroupTot}
      })
    }

    callParentGiveSum(sumGroupTot, sumGroupTotTip){
      this.props.parentReference(sumGroupTot, sumGroupTotTip)
    }

  handleInputFocus = (item, index) => {
    this.setState({selectedItemIndex: index})
  }

  handleGroupFocus = (item, index) => {
    this.setState({selectedGroupIndex: index})
  }

  addPerson(index) {
    let groupIdx = index
    this.setState(prevState => {
      let groupies = [...prevState.Group]
      let prevNumberPersons = groupies[groupIdx].numberPersons
      groupies[groupIdx] = {...groupies[groupIdx], numberPersons: (prevNumberPersons+1)}

      return {Group: groupies}
    })
  }

  increasePercentage(index) {
    let groupIdx = index
    this.setState(prevState => {
      let groupies = [...prevState.Group]
      let prevPercentage = groupies[groupIdx].tipPercentage

      groupies[groupIdx] = {...groupies[groupIdx], tipPercentage: (prevPercentage+1)}

      let groupTotals = groupies.map(group => ({
        key: group.key,
        groupTot: group.groupTotal,
        groupTotTip: group.tipPercentage
      }))

      let sumGroupTotalTips = groupTotals.reduce(function (accumulator, currentValue) {
        return Math.round( ((parseFloat(accumulator) + parseFloat(currentValue.groupTot*(currentValue.groupTotTip/100))) + Number.EPSILON) * 100) / 100
      }, 0)

      this.callParentGiveSum(-1, sumGroupTotalTips)

      return {Group: groupies}
    })
  }

  decreasePercentage(index) {
    let groupIdx = index
    this.setState(prevState => {
      let groupies = [...prevState.Group]
      let prevPercentage = groupies[groupIdx].tipPercentage
      groupies[groupIdx] = {...groupies[groupIdx], tipPercentage: (prevPercentage-1)}

      let groupTotals = groupies.map(group => ({
        key: group.key,
        groupTot: group.groupTotal,
        groupTotTip: group.tipPercentage
      }))

      let sumGroupTotalTips = groupTotals.reduce(function (accumulator, currentValue) {
        return Math.round( ((parseFloat(accumulator) + parseFloat(currentValue.groupTot*(currentValue.groupTotTip/100))) + Number.EPSILON) * 100) / 100
      }, 0)

      this.callParentGiveSum(-1, sumGroupTotalTips)

      return {Group: groupies}
    })
  }

  removePerson(index) {
    let groupIdx = index
    this.setState(prevState => {
      let groupies = [...prevState.Group]
      let prevNumberPersons = groupies[groupIdx].numberPersons
      groupies[groupIdx] = {...groupies[groupIdx], numberPersons: (prevNumberPersons-1)}

      return {Group: groupies}
    })
  }

  updateIndex = () => {
  }

  sendInput(inputText) {
    let groupIdx = this.state.selectedGroupIndex

    this.setState({isDialogVisible: false})
    this.setState(prevState => {
      let groupies = [...prevState.Group]
      groupies[groupIdx] = {...groupies[groupIdx], title: inputText}

      return {Group: groupies}
    })

  }

  handleEditGroupName(item, index) {
    this.setState({selectedGroupIndex: index})
    this.setState({isDialogVisible: true})
  }

  renderGroupItemInput = ({item, index}) => {
    return (
      <View style={styles.itemBox}>
        <View style={styles.inlineContainer}>
          <View style={{width: '65%', flexDirection: 'row'}}>

            <Text style={styles.currencySymbol}>£</Text>
            <Input
              ref={ref => { this.input = ref }}
              placeholder={'0.00'}
              inputStyle={styles.itemAmountBox}
              placeholderTextColor="#000"
              keyboardType='numeric'
              onFocus={() => this.handleInputFocus(item, index)}
              onChangeText={item => this.handleOnChangeText(item)}
            />
          </View>

          <View style={{width: '35%'}}>
            <ItemTypePicker/>

          </View>
        </View>
      </View>
    )
  }

  renderGroupItem = ({item, index}) => {
    return (
      <View style={styles.mainView} >
        <View style={styles.titleBox}>
          <View style={{width: 100}}>
            <Icon
                size={23}
                name='trash'
                type='font-awesome'
                color='#cc6666'
                onPress={() => this.handleDeleteGroup(item, index)}
                iconStyle={{padding: 5, paddingHorizontal: 10, alignSelf: 'flex-start'}}
            />
          </View>
          <View style={{flexDirection: 'row'}}>
            <Text style={styles.itemTitle}> {item.title} </Text>
            <Icon
                size={22}
                name='edit'
                type='font-awesome'
                color='#bab6b3'
                iconStyle={{paddingTop: 8}}
                onPress={() => this.handleEditGroupName(item, index)}
            />
          </View>
          <View style={{width: 100}}>
          </View>
        </View>

        <View style={styles.groupBox}>
          <FlatList
              keyboardDismissMode='on-drag'
              onFocus={() => this.handleGroupFocus(item, index)}
              data={item.billItem}
              renderItem = {this.renderGroupItemInput}
              keyExtractor={(item, index) => 'item.id'+index}
          />
          <View style={{alignItems: 'center'}}>
            <Text style={{fontSize: 18}}>£{item.groupTotal}</Text>
          </View>
          <View style={styles.groupTotalBox}>
            <View style={{justifyContent: 'space-between', flexDirection: 'row'}}>
              <View style={{flexDirection: 'row'}}>
                <Icon
                    size={20}
                    name='remove'
                    color='red'
                    onPress={()=>this.removePerson(index)}
                />
                <Text>{item.numberPersons}</Text>
                <Icon
                    size={20}
                    name='person'
                    color='grey'
                />
                <Icon
                    size={20}
                    name='add'
                    color='green'
                    onPress={()=>this.addPerson(index)}
                />
              </View>
              <View style={{flexDirection: 'column'}}>
                <View style={{flexDirection: 'row'}}>
                  <Icon
                      size={20}
                      name='remove'
                      color='red'
                      onPress={()=>this.decreasePercentage(index)}
                  />
                  <Text>{item.tipPercentage} %</Text>
                  <Icon
                      size={20}
                      name='add'
                      color='green'
                      onPress={()=>this.increasePercentage(index)}
                  />
                </View>
                <View style={{alignItems: 'flex-end'}}>
                  <Text>+ £{Math.round( ((item.groupTotal * (item.tipPercentage/100)) + Number.EPSILON) * 100 )/ 100}</Text>
                </View>
              </View>
            </View>

            <View style={{alignItems: 'center'}}>
              <Text style={{fontSize: 18}}>£{Math.round( (( (item.groupTotal/item.numberPersons) * (1+ (item.tipPercentage/100) ) ) + Number.EPSILON) * 100 )/ 100 } each</Text>
            </View>

          </View>
        </View>
      </View>
    )
  }

  render() {
    return (
        <FlatList
          keyboardDismissMode='on-drag'
          keyboardShouldPersistTaps='always'
          data={this.state.Group}
          width='100%'

          ref={(ref) => {
            this.groupList_Ref = ref;
          }}

          keyExtractor={(item, index) => 'item.key'+index}
          renderItem = {this.renderGroupItem}
          extraData={this.state}
          ListHeaderComponent={
            <DialogInput isDialogVisible={this.state.isDialogVisible}
              title={"Edit group name"}
              message={"Enter a new name for the group"}
              hintInput ={"Group " + this.state.selectedGroupIndex}
              submitInput={ (inputText) => {this.sendInput(inputText)} }
              closeDialog={ () => {this.setState({isDialogVisible: false})}}>
            </DialogInput>
          }
          ListFooterComponent={<View style={styles.addGroupContainer}><Icon
                                  raised
                                  name='plus'
                                  type='font-awesome'
                                  color='#a3c1ad'
                                  onPress={() => this.newGroup()}
                                /></View>
          }
        />

    );
  }
}

const styles = StyleSheet.create({
  mainView: {
    marginTop: 30,
  },
  currencySymbol: {
    fontSize: 18,
    marginTop: 8,
  },
  billTotalFooter: {
    fontSize: 20,
    alignItems: 'flex-start',
    backgroundColor: '#fff',
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: -1,
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.00,

    elevation: 3,
  },
  addGroupContainer:{
    alignItems: 'center',
    paddingTop: 15,
    marginBottom: 50,
  },
  groupTotalBox: {
    backgroundColor: '#bfc1c2',
    paddingVertical: 10,
    flexDirection: 'column',
    paddingHorizontal: 20,
  },
  groupBox: {
    width: '80%',
    alignSelf: 'center',
    backgroundColor: '#ededed',
    borderRadius: 1.8,
    shadowColor: "#000",
    shadowOffset: {
    	width: 0,
    	height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.00,

    elevation: 3,
  },
  inlineContainer: {
    flexDirection: 'row',
    paddingHorizontal: 10,
  },
  itemBox: {
    backgroundColor: '#ededed',
    marginLeft: 10,
  },
  titleBox: {
    width: '90%',
    alignSelf: 'center',
    backgroundColor: '#3A4A4D',
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderRadius: 3.6,
    shadowColor: "#000",
    shadowOffset: {
    	width: 0,
    	height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,

    elevation: 5,
  },
  itemTitle: {
    fontSize: 20,
    padding: 4,
    color: '#bfc1c2',
  },
  itemIconBox: {
    marginTop: 12,
    paddingHorizontal: 5,
  },
  itemAmountBox: {
    color: '#424142',
  },
});
