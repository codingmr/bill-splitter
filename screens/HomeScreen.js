import React, { Component } from 'react';
import { Image, FlatList, Alert, Platform, ListItem, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';
import * as WebBrowser from 'expo-web-browser';


import { Icon, Input } from 'react-native-elements'


import { MonoText } from '../components/StyledText';

import GroupList from '../components/GroupList';

export default class HomeScreen extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      textInput_Holder: '',
      count: 3,
      Group: [
        {
          title: 'Group 1',
          billItem: [{itemAmount: '£1.20', itemIcon: 'restaurant'}, {itemAmount: '£3.50', itemIcon: 'restaurant'}],
        },
        {
          title: 'Group 2',
          billItem: [{itemAmount: '£10.20', itemIcon: 'restaurant'}, {itemAmount: '£2.99', itemIcon: 'restaurant'}],
        }
      ]
    }

  }

  joinData = () => {
    this.setState({'count': this.state.count + 1})

    this.setState({ Group: [...this.state.Group, ...[{title: 'Group ' + this.state.count, billItem: [{itemAmount: '£0.00', itemIcon: 'restaurant'}]}]] })
  }

  FlatListItemSeparator = () => {
    return (
      <View
        style={{
          height: 20,
          width: "100%",
        }}
      />
    );
  }

  GetItem(item) {

    Alert.alert(item);

  }

  renderGroupItem = ({item}) => {
    return (
      <View>
        <View style={styles.itemBox}>
          <Text style={styles.itemTitle} onPress={this.GetItem.bind(this, item.title)}> {item.title} </Text>
        </View>
        <FlatList
            data={item.billItem}
            renderItem = {({ item }) => <View style={styles.groupBox}>
                                          <View style={styles.inlineContainer}>
                                            <Input
                                              placeholder={item.itemAmount}
                                              inputStyle={styles.itemAmountBox}
                                            />
                                            <Icon
                                              name={item.itemIcon}
                                              type='Ionicon'
                                              iconStyle={styles.itemIconBox}
                                            />
                                          </View>
                                        </View>}

            listKey={(item, index) => 'D' + index.toString()}
        />
      </View>
    )
  }

  render() {
    return (
      <View style={styles.container}>
        <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>

          <View style={styles.getStartedContainer}>

            <Text style={styles.getStartedText}>
              Welcome! This is Bill Splitter. A quick way to split the bill among friends.
            </Text>
          </View>

          <FlatList
            data={this.state.Group}
            width='100%'
            listKey={(item, index) => 'D' + index.toString()}
            ItemSeparatorComponent={this.FlatListItemSeparator}
            renderItem = {this.renderGroupItem}
          />

          <View style={styles.addGroupContainer}>

            <Icon
              raised
              name='plus'
              type='font-awesome'
              color='#a3c1ad'
              onPress={this.joinData} />
          </View>

        </ScrollView>



      </View>
    );
  }
}

HomeScreen.navigationOptions = {
  header: null,
};

function handleExample() {
  //do something
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  addGroupContainer:{
    paddingTop: 5,
    alignItems: 'center',
  },
  developmentModeText: {
    marginBottom: 20,
    color: 'rgba(0,0,0,0.4)',
    fontSize: 14,
    lineHeight: 19,
    textAlign: 'center',
  },
  groupBox: {
    backgroundColor: 'blue',
    width: '80%',
    alignSelf: 'center',
    marginBottom: 10,
  },
  inlineContainer: {
    backgroundColor: 'red',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 50,
    paddingTop: 5,
    marginBottom: 5,
  },
  itemBox: {
    width: '80%',
    alignSelf: 'center',
    backgroundColor: 'green',
    alignItems: 'center',
  },
  itemTitle: {
    fontSize: 24,
  },
  itemIconBox: {
    marginTop: 12,
  },
  itemAmountBox: {
  },
  contentContainer: {
    paddingTop: 30,
  },
  welcomeContainer: {
    alignItems: 'center',
    marginTop: 10,
    marginBottom: 20,
  },
  welcomeImage: {
    width: 100,
    height: 80,
    resizeMode: 'contain',
    marginTop: 3,
    marginLeft: -10,
  },
  getStartedContainer: {
    alignItems: 'center',
    marginHorizontal: 50,
  },
  homeScreenFilename: {
    marginVertical: 7,
  },
  codeHighlightText: {
    color: 'rgba(96,100,109, 0.8)',
  },
  codeHighlightContainer: {
    backgroundColor: 'rgba(0,0,0,0.05)',
    borderRadius: 3,
    paddingHorizontal: 4,
  },
  getStartedText: {
    fontSize: 17,
    color: 'rgba(96,100,109, 1)',
    lineHeight: 24,
    textAlign: 'center',
  },
  tabBarInfoContainer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { width: 0, height: -3 },
        shadowOpacity: 0.1,
        shadowRadius: 3,
      },
      android: {
        elevation: 20,
      },
    }),
    alignItems: 'center',
    backgroundColor: '#fbfbfb',
    paddingVertical: 20,
  },
  tabBarInfoText: {
    fontSize: 17,
    color: 'rgba(96,100,109, 1)',
    textAlign: 'center',
  },
  navigationFilename: {
    marginTop: 5,
  },
  helpContainer: {
    marginTop: 15,
    alignItems: 'center',
  },
  helpLink: {
    paddingVertical: 15,
  },
  helpLinkText: {
    fontSize: 14,
    color: '#2e78b7',
  },
});
