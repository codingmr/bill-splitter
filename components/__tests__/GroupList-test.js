import React from 'react';
import GroupList from '../GroupList';
import HomeScreen from '../../screens/HomeScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<GroupList />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('New group was added correctly', ()=>{
  let GroupData=renderer.create(<GroupList/>).getInstance();
  GroupData.newGroup()

  if (expect(GroupData.state.Group.length).toEqual(1)) {
    expect(GroupData.state.Group[0].title).toEqual("Group 1")
    if (expect(GroupData.state.Group[0].billItem.length).toEqual(1)) {
      expect(GroupData.state.Group[0].billItem[0].toEqual({id: 0, itemAmount: '0.00', itemIcon: 'restaurant'}))
    }
  }
})
/*
it('Bill total was updated correctly', ()=>{
  let GroupData=renderer.create(<GroupList/>).getInstance();

  let testGroupData = [
    {
      key: 1,
      title: "Group 1",
      groupTotal: 0,
      billItem: [
        {
          id: 0,
          itemAmount: "11.91",
          itemIcon: "restaurant"
        },
        {
          id: 1,
          itemAmount: "12.35",
          itemIcon: "restaurant"
        },
        {
          id: 2,
          itemAmount: "2.57",
          itemIcon: "restaurant"
        }
      ]
    },
    {
      key: 2,
      title: "Group 2",
      groupTotal: 0,
      billItem: [
        {
          id: 0,
          itemAmount: "5.33",
          itemIcon: "restaurant"
        },
        {
          id: 1,
          itemAmount: "1.72",
          itemIcon: "restaurant"
        }
      ]
    }
  ]

  GroupData.state.Group = testGroupData

  GroupData.updateBillTotal()

  console.log(JSON.stringify(GroupData.state.Group, null, 1))

  expect(GroupData.state.billTotal).toEqual(33.88)

})

*/

// New group was added

// Group was deleted

// Group was editted
