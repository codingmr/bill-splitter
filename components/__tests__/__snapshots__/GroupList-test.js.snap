// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  ListFooterComponent={
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 50,
          "paddingTop": 15,
        }
      }
    >
      <Themed.Icon
        color="#a3c1ad"
        name="plus"
        onPress={[Function]}
        raised={true}
        type="font-awesome"
      />
    </View>
  }
  ListHeaderComponent={
    <DialogInput
      closeDialog={[Function]}
      hintInput="Group -1"
      isDialogVisible={false}
      message="Enter a new name for the group"
      submitInput={[Function]}
      title="Edit group name"
    />
  }
  data={Array []}
  disableVirtualization={false}
  extraData={
    Object {
      "Group": Array [],
      "billTotal": 0,
      "count": 0,
      "isDialogVisible": false,
      "selectedGroupIndex": -1,
      "selectedItemIndex": -1,
      "textHolder": "",
    }
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  keyboardDismissMode="on-drag"
  keyboardShouldPersistTaps="always"
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  width="100%"
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#E3E6E7",
                    "borderRadius": 10,
                    "marginLeft": 30,
                    "marginRight": 30,
                    "minWidth": 300,
                  },
                  Object {},
                ]
              }
            >
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "marginBottom": 5,
                      "marginTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  Edit group name
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "marginBottom": 10,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Enter a new name for the group
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={true}
                  autoFocus={true}
                  clearButtonMode="never"
                  clearTextOnFocus={false}
                  keyboardType="default"
                  onChangeText={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Group -1"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#B0B0B0",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "color": "rgba(0,0,0,0.54)",
                      "fontSize": 16,
                      "marginBottom": 15,
                      "marginTop": 10,
                      "paddingBottom": 5,
                      "paddingLeft": 10,
                      "paddingTop": 5,
                      "textAlign": "left",
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "borderColor": "#B0B0B0",
                    "borderTopWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "maxHeight": 48,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "borderColor": "#B0B0B0",
                        "borderRightWidth": 5,
                        "color": "#408AE2",
                        "fontSize": 18,
                        "height": 48,
                        "maxHeight": 48,
                        "padding": 10,
                        "textAlign": "center",
                      }
                    }
                  >
                    Cancel
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#B0B0B0",
                      "width": 1,
                    }
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#408AE2",
                        "fontSize": 18,
                        "padding": 10,
                        "textAlign": "center",
                      }
                    }
                  >
                    Submit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 50,
            "paddingTop": 15,
          }
        }
      >
        <View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderRadius": 28,
                "height": 52,
                "justifyContent": "center",
                "margin": 7,
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
                "width": 52,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
